UefiPayloadPkg/UefiPayloadEntry/AcpiTable.c

```diff
@@ -128,11 +128,11 @@ ParseAcpiInfo (
   //
   // Verify values for proper operation
   //
-  ASSERT (Fadt->Pm1aCntBlk != 0);
-  ASSERT (Fadt->PmTmrBlk != 0);
-  ASSERT (Fadt->ResetReg.Address != 0);
-  ASSERT (Fadt->Pm1aEvtBlk != 0);
-  ASSERT (Fadt->Gpe0Blk != 0);
+  // ASSERT (Fadt->Pm1aCntBlk != 0);
+  // ASSERT (Fadt->PmTmrBlk != 0);
+  // ASSERT (Fadt->ResetReg.Address != 0);
+  // ASSERT (Fadt->Pm1aEvtBlk != 0);
+  // ASSERT (Fadt->Gpe0Blk != 0);
 

```

Comment:
These platform specific information is consumed by other drivers. e.g.: PmTimerRegBase (Fadt->PmTmrBlk) is used by AcpiTimerLib. If it is not set correctly by bootloader and we skipped the check here, it might cause issue and spend more time to debug and fix.  Is there any reason why bootloader doesn't fill them following ACPI spec?

Reply:
AFAIK none of these are mandatory fields and while for one platform it maybe true for another it may not. We can not impose hard requirement in the common path. This was introduced before UPL 2.0 so I think it requires some refactoring (maybe pull in such info in Arch specific area etc). So for now this is just a quick fix WA with a TODO

Reply:
This is EDKII implementation choice since these information are required by UEFI payload, but it is not defined by universal payload spec. The implementation just gets these information from existing standards. If there are other methods to get these information, we could revisit this I think.

Reply:
ACPI spec has defined that those HW register interface are optional and can be ignored by ACPI OS:
https://uefi.org/specs/ACPI/6.5/05_ACPI_Software_Programming_Model.html#fixed-acpi-description-table-fadt

I would recommend that UPL does not make them as mandatory for all the platforms.

@dhaval-rivos , could you help to keep those ASSERT there in this PR? We will have another PR later to move those ASSERT to consumer library code so it is mandatory only when they are really consumed by UPL.

Reply:
I could do that but the issue is, with those asserts my platform would stop booting (in debug). On the other hand, just a commented out WA with TODO, will not break anything on your end. So we can keep both our systems going that way. Let future patch fix it properly. Does that work?

Reply:
ok to me.

Reply:
ok. once this PR is merged, will create a new PR to update it.